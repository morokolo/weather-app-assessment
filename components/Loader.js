import React from "react";
import {
  ActivityIndicator,
  Platform,
  SafeAreaView,
  StatusBar,
  StyleSheet,
} from "react-native";

export default function Loader() {
  return (
    <SafeAreaView style={styles.loading}>
      <ActivityIndicator size="large" color="#fc5c65" />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  loading: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
    paddingTop: Platform.OS === "android" ? StatusBar.currentHeight : 0,
  },
});
