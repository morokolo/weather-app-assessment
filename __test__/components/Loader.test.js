import "react-native";
import React from "react";
import Loader from "../../components/Loader";
import renderer from "react-test-renderer";

it("should render the App component", () => {
  const tree = renderer.create(<Loader />).toJSON();

  expect(tree).toMatchSnapshot();
});
