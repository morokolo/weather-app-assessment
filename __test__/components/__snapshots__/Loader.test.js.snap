// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the App component 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingTop": 0,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#fc5c65"
    hidesWhenStopped={true}
    size="large"
  />
</RCTSafeAreaView>
`;
